src/devices/Bno055/Bno055Sensor.cs

```diff
@@ -548,17 +536,14 @@ private void SetConfigMode(bool mode)
         /// <inheritdoc/>
         public void Dispose()
         {
-            if (_autoDispoable)
+            if (_shouldDispose)
             {
                 _i2cDevice?.Dispose();
                 _i2cDevice = null!;
```

Comment:
Like other places, just because we're not `_shouldDispose` does that mean we should still keep the ref live in `_i2cDevice` or should we be nulling it outside the brace?